addons/website/static/src/js/shared_blocks.js

```diff
@@ -0,0 +1,63 @@
+/** @odoo-module **/
+
+import publicWidget from 'web.public.widget';
+
+publicWidget.registry.o_shared_blocks = publicWidget.Widget.extend({
+    selector: '#o_shared_blocks',
+    disabledInEditableMode: false,
+    events: {
+        'show.bs.modal': '_onContentChanged',
+        'hidden.bs.modal': '_onContentChanged',
+    },
+    edit_events: {
+        'content_changed': '_onContentChanged',
+    },
+
+    /**
+     * @override
+     */
+    start() {
+        this._updateTargetVisibility();
+        return this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this.el.classList.add('d-none');
+        this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @returns {boolean}
+     */
+    _hasVisibleChild() {
+        return !!this.el.querySelector(':scope > *:not(.d-none)');
+    },
+    /**
+     * @private
+     */
+    _updateTargetVisibility() {
+        if (this._hasVisibleChild()) {
+            this.el.classList.remove('d-none');
+        } else {
+            this.el.classList.add('d-none');
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _onContentChanged() {
+        this._updateTargetVisibility();
+    },
+});
```

Comment:
Not a 100% sure but I probably told you that if possible that shared div should be hidden if nothing is in it? That's true ideally... but this is overkill. 63 lines of JS for a micro improvement is not worth it, so:
- Are you sure that the same result cannot be achieved in one line of CSS ? I would have assumed that `#o_shared_blocks:blank { display: none; }` or at worst `#o_shared_blocks:empty { display: none; }` would have done the trick (I know that `:blank` may not be well supported, at least in the past).
- If not, then let's forget about it, let's just always show that div... unless there are problems when showing it in edit mode?

Reply:
> I probably told you

Yes :+1: 

The `:blank` selects empty user input elements so that does not work for this case and `:empty` selects any element that has no children so when there is an all pages pop up it does not work.
Since we only use the new area for popup we could use something like that:
`body:not(.modal-open) #o_shared_blocks {display: none;}` but as soon as we will add the ability to have other blocks shared on all pages we will have to find another solution.. So I delete the public widget, and I can add `body:not(.modal-open) #o_shared_blocks {display: none;}` until we allow to put another block in the shared blocks area, or you think it's useless ?

Reply:
@Guillaume-gdi 

Missed this message :grimacing: 
Go with `#o_shared_blocks:empty {display: none;}` (hidden when no modal on all pages (most websites) and visisble when there could be one, open or not).

Also rebase the branch and remove the draft once you did that :+1:

Reply:
EDIT: I see there are still temporary commits on this, can you squash what needs to be squashed etc ? :)